### DISCUSSION TITLE
Growable Buffer: baseline performance tests

### DISCUSSION THREAD

**Discussion by ianna** [scikit-hep/awkward#2362] [General]
![https://user-images.githubusercontent.com/1390682/230147717-453f1912-7587-4f37-a333-6f1d49cbdb76.jpeg](https://github.com/scikit-hep/awkward/files/11161042/30A48871-9569-4BD9-AD77-B1A2CF23CECA.pdf)
![https://user-images.githubusercontent.com/1390682/230148000-6456a325-3a10-432a-9a12-2d8b3b40a13a.jpeg](https://github.com/scikit-hep/awkward/files/11161191/E8F925AA-0DF0-4724-8663-C0347925E8B0.pdf)

**Comment by ianna** [comment]
@jpivarski - as discussed, here are the plots.

**Comment by jpivarski** [comment]
The PDFs will be useful for talks, but for looking up this page, it will help to have large PNGs.

<img src="https://user-images.githubusercontent.com/1852447/230164867-88a7f6c5-aa92-4293-9028-326a035d4b88.png" width="100%">

<img src="https://user-images.githubusercontent.com/1852447/230164964-9bded2b3-7288-4a0c-a695-cc74e1637d40.png" width="100%">

Words to include here so that we can find them later: GrowableBuffer panel panels std::vector like exponential memory growth versus linear speed time seconds performance allocation memcpy malloc calloc
